### PR TITLE
14574 implement header footer

### PR DESCRIPTION
**Pull request by jtmst**
## Description
Adds header and footer to next-build
Renders nav components and mega menu.  Not all content-build functionality is ported over.
Data is sourced from a JSON object in importHeaderData.js.  This data was pulled from the prod JSON file

Relates to #14574
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14574

## Testing done
Local visual testing

## Screenshots
Screen Shot 2023-08-14 at 10 28 46 AM
Screen Shot 2023-08-14 at 10 28 37 AM
Screen Shot 2023-08-14 at 10 30 54 AM

## QA steps

What needs to be checked to prove this works?  
Navigate to any existing page in next build and verify that the header and footer are rendered

What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?

As user navigating va.gov

1. Do this
    - [ ] Validate that the header renders
2. Then
    - [ ] Validate that the footer renders
3. Then validate Acceptance Criteria from issue
    - [ ]  Header and footer menus render when a Next Build static page is viewed
    - [ ] Data is accurate and sourced from a current headerFooter.json and written to page
  
### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### CODE REVIEW

**Comment by tjheffner** [comment]
looks like couple lint errors related to `` tags in `TopNav.tsx`
```
./src/templates/globals/header/TopNav.tsx
10:19  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
35:19  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
45:74  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
51:19  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
59:76  Error: `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.  react/no-unescaped-entities
79:15  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
97:13  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element
```

I will look at why tugboat didn't auto-build this PR.

File: src/templates/globals/wrapper/index.tsx [hunk 1]
```diff
@@ -40,8 +41,10 @@ export function Wrapper({ bannerData, children }: LayoutProps) {
 
   return (
     <>
-      {/* {showBanners ? mapBanners : null} */}
+      {showBanners ? mapBanners : null}
```
Comment (tjheffner) [inline src/templates/globals/wrapper/index.tsx:44]: Will you move these banners below the header?
looking at https://pr141-yb1aiq9d4a369sojvmry5r2rman1gavn.tugboat.vfs.va.gov/butler-health-care/stories/ vs https://www.va.gov/butler-health-care/stories/

File: src/pages/_document.tsx [hunk 2]
```diff
@@ -1,36 +1,47 @@
 import NextDocument, { Html, Main, NextScript, Head } from 'next/document'
```
Comment (tjheffner) [inline src/pages/_document.tsx:1]: do we need to import NextDocument still?

File: src/pages/_document.tsx [hunk 3]
```diff
@@ -1,36 +1,47 @@
...
+          rel="stylesheet"
+          data-entry-name="style.css"
+          href="https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com/generated/style.css"
+        />
+        <Script
+          id="staticPages"
+          strategy="afterInteractive"
+          src="https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com/generated/static-pages.entry.js"
+        />
+        <Script
```
Comment (tjheffner) [inline src/pages/_document.tsx:25]: will you swap the order of these two and/or add a comment explaining that the import order here matters, because we need the data added to the window object (or whatever the reason actually is)

File: public/importHeaderData.js [hunk 4]
```diff
@@ -0,0 +1,1028 @@
+const headerData = {
```
Comment (tjheffner) [inline public/importHeaderData.js:1]: will you add a comment to this file also explaining how it interacts with the static pages entry script. I am mostly thinking for the part where we co-exist in production with content-build, we will likely use that generated data file, but we'll still need the lines in here to add it to the window object

File: src/templates/globals/footer/staticData/footer-links.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+export const footerLinks = [
```
Comment (tjheffner) [inline src/templates/globals/footer/staticData/footer-links.js:1]: why is this file necessary if these items all exist in `importHeaderData.js` ?

**Review by tjheffner — CHANGES_REQUESTED** [review]
overall, looks super. thank you for tackling this

one small nit, some requests for comments and some general questions

File: src/templates/globals/wrapper/index.tsx (hunk 1 above)
Comment (jtmst) [reply src/templates/globals/wrapper/index.tsx:44]: good call, will fix

File: src/pages/_document.tsx (hunk 2 above)
Comment (jtmst) [reply src/pages/_document.tsx:1]: We do not, will remove

File: public/importHeaderData.js (hunk 4 above)
Comment (jtmst) [reply public/importHeaderData.js:1]: Added a comment with code to clarify

File: src/templates/globals/footer/staticData/footer-links.js (hunk 5 above)
Comment (jtmst) [reply src/templates/globals/footer/staticData/footer-links.js:1]: Oversite on my part, and this makes me see there is a good deal of other footer related code I can clean up as well

**Review by tjheffner — APPROVED** [review]
